apps/desktop/src/lib/selection/entityAdapters.ts

```diff
@@ -0,0 +1,52 @@
+import { createEntityAdapter } from '@reduxjs/toolkit';
+import type { TreeChange } from '$lib/hunks/change';
+import type { HunkAssignment, HunkHeader } from '$lib/hunks/hunk';
+import type { LineId } from '@gitbutler/ui/utils/diffParsing';
+
+/**
+ * Assignments and selections are keyed by this combination of parameters.
+ */
+export function compositeKey(args: {
+	stackId: string | null;
+	path: string;
+	hunkHeader: string | HunkHeader | null;
+}) {
+	if (typeof args.hunkHeader === 'string' || args.hunkHeader === null) {
+		return `${args.stackId}::${args.path}::${args.hunkHeader}`;
+	}
+	return `${args.stackId}::${args.path}::${args.hunkHeader?.newStart || null}`;
+}
+
+export const treeChangeAdapter = createEntityAdapter<TreeChange, string>({
+	selectId: (change) => change.path,
+	sortComparer: (a, b) => sortLikeLs(a.path, b.path)
```

Comment:
Is the sorting of these ever used? Don't we re-do the sorting in our file view components?